services-camel/src/main/java/edu/unc/lib/cdr/ReplicationProcessor.java

```diff
@@ -0,0 +1,154 @@
+package edu.unc.lib.cdr;
+
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryChecksum;
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryPath;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringJoiner;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+
+public class ReplicationProcessor implements Processor {
+	private static final Logger log = LoggerFactory.getLogger(ReplicationProcessor.class);
+	
+	private final String replicationLocations;
+	private final int maxRetries;
+	private final long retryDelay;
+	
+	public ReplicationProcessor(String replicationLocations, int maxRetries, long retryDelay) {
+		this.replicationLocations = replicationLocations;
+		this.maxRetries = maxRetries;
+		this.retryDelay = retryDelay;
+		
+		checkReplicationLocations(splitReplicationLocations(replicationLocations));
+	}
+	
+	@Override
+	public void process(Exchange exchange) throws Exception {
+		final Message in = exchange.getIn();
+		
+		String binaryPath = (String) in.getHeader(CdrBinaryPath);
+		String binaryChecksum = (String) in.getHeader(CdrBinaryChecksum);
+		String[] replicationPaths = splitReplicationLocations(replicationLocations);
+		
+		int retryAttempt = 0;
+		
+		while (true) {
+			try {
+				replicate(binaryPath, binaryChecksum, replicationPaths);
+				break;
+			} catch (Exception e) {
+				if (retryAttempt == maxRetries) {
+					throw e;
+				}
+
+				retryAttempt++;
+				TimeUnit.MILLISECONDS.sleep(retryDelay);
+			}
+		}
+	}
+	
+	private String[] splitReplicationLocations(String replicationLocations) {
+		return replicationLocations.split(";");
+	}
+	
+	private boolean checkReplicationLocations(String[] replicationPaths) {
+		for (String replicationPath : replicationPaths) {
+			if (!Files.exists(Paths.get(replicationPath))) {
+				throw new ReplicationDestinationUnavailableException("Unable to find replication destination {}", replicationPath);
+			}
+		}
+		
+		return true;
+	}
+	
+	private String createFilePath(String basePath, String originalFileChecksum) {
+		String[] binaryFcrepoChecksumSplit = originalFileChecksum.split(":");
+		String checksum = binaryFcrepoChecksumSplit[2];
+		
+		String[] tokens = Iterables.toArray
+				(Splitter.fixedLength(2).split(checksum), 
+						String.class);
+		
+		String remotePath = new StringJoiner("/")
```

Comment:
The files in my VM have this sort of path /var/lib/tomcat7/fcrepo4-data/fcrepo.binary.directory/02/f1/c2/02f1c2f26b4adc0e23bef86ae481b5e81c2d99afp